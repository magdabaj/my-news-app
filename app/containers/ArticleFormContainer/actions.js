/*
 *
 * ArticleFormContainer actions
 *
 */

import {ADD_ARTICLE, ADD_ARTICLE_CANCELED, ADD_ARTICLE_ERROR, ADD_ARTICLE_SUCCESS} from "./constants";

export const addArticle = (article, tagId, user) => ({
  type: ADD_ARTICLE,
  article,
  tagId,
  user
});

export const addArticleSuccess = article => ({
  type: ADD_ARTICLE_SUCCESS,
  article,
});

export const addArticleError = error => ({
  type: ADD_ARTICLE_ERROR,
  error,
});

export const addArticleCanceled = () => ({
  type: ADD_ARTICLE_CANCELED,
});