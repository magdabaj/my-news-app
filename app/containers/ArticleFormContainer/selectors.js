import { createSelector } from "reselect";
import { initialState } from "./reducer";

/**
 * Direct selector to the articleFormContainer state domain
 */

const selectArticleFormContainerDomain = state =>
  state.articleFormContainer || initialState;

/**
 * Other specific selectors
 */

/**
 * Default selector used by ArticleFormContainer
 */

const makeSelectArticleFormContainer = () =>
  createSelector(
    selectArticleFormContainerDomain,
    substate => substate
  );

export default makeSelectArticleFormContainer;
export { selectArticleFormContainerDomain };
