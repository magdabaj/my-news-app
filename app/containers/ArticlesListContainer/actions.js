/*
 *
 * ArticlesListContainer actions
 *
 */

import {
  GET_ALL_ARTICLES,
  GET_ARTICLES_FOR_TAG,
  GET_ARTICLES_FOR_TAG_FAILED,
  GET_ARTICLES_FOR_TAG_SUCCESS, START_ADD
} from "./constants";

export const getAllArticles = () => ({
  type: GET_ALL_ARTICLES
});

export const getArticlesForTag = tagId => ({
  type: GET_ARTICLES_FOR_TAG,
  tagId,
});

export const getArticlesForTagSuccess = articles => ({
  type: GET_ARTICLES_FOR_TAG_SUCCESS,
  articles,
});

export const getArticlesForTagFailed = error => ({
  type: GET_ARTICLES_FOR_TAG_FAILED,
  error,
});

export const startAdd = tagName => ({
  type: START_ADD,
  tagName,
});