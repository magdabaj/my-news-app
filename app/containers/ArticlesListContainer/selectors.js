import { createSelector } from "reselect";
import { initialState } from "./reducer";
import makeSelectNavigationContainer, {selectNavigationContainerDomain} from "../NavigationContainer/selectors";

/**
 * Direct selector to the articlesListContainer state domain
 */

const selectArticlesListContainerDomain = state =>
  state.articlesListContainer || initialState;

const selectArticlesRoute = (state, props) =>
    props.match.params.tag;

const selectSlug = () =>
    createSelector(
        selectArticlesRoute,
        tag => tag,
    );

const selectTag = () =>
    createSelector(
        makeSelectNavigationContainer(),
        selectArticlesRoute,
        (navigationState, routeArticleName) => {
            const selectedTag = navigationState.tags.find(tag => tag.name === routeArticleName);

            return selectedTag || {
                name: '',
                id: null,
            }
        }
    );

/**
 * Other specific selectors
 */

/**
 * Default selector used by ArticlesListContainer
 */

const makeSelectArticlesListContainer = () =>
  createSelector(
    selectArticlesListContainerDomain,
    substate => substate,
  );

const makeSelectArticles = () =>
    createSelector(
        selectArticlesListContainerDomain,
        substate => substate.articlesForTag,
    );

export default makeSelectArticlesListContainer;
export { selectArticlesListContainerDomain, selectTag, makeSelectArticles, selectSlug };
