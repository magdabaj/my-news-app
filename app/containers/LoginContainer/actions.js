/*
 *
 * LoginContainer actions
 *
 */

import {CANCEL_LOGIN, LOGIN, LOGIN_FAILED, LOGIN_SUCCESS} from "./constants";

export const login = (email, password) => ({
  type: LOGIN,
  email,
  password,
});

export const loginSuccess = user => ({
  type: LOGIN_SUCCESS,
  user,
});

export const loginFailed = error => ({
  type: LOGIN_FAILED,
  error,
});

export const cancelLogin = () => ({
  type: CANCEL_LOGIN,
});