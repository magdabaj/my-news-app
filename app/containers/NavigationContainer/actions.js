/*
 *
 * NavigationContainer actions
 *
 */

import { GET_TAGS, GET_ARTICLES_BY_TAG, GET_TAGS_FAILED, GET_TAGS_SUCCESS } from "./constants";

export const getTags = () => ({
  type: GET_TAGS
});

export const getTagsSuccess = tags => ({
  type: GET_TAGS_SUCCESS,
  tags
});

export const getTagsFailed = error => ({
  type: GET_TAGS_FAILED,
  error
});

export const getArticlesByTag = tagId => ({
  type: GET_ARTICLES_BY_TAG,
  tagId,
});
