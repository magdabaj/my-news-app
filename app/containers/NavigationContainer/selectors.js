import { createSelector } from "reselect";
import { initialState } from "./reducer";

/**
 * Direct selector to the navigationContainer state domain
 */

const selectNavigationContainerDomain = state =>
  state.navigationContainer || initialState;

/**
 * Other specific selectors
 */

/**
 * Default selector used by NavigationContainer
 */

const makeSelectNavigationContainer = () =>
  createSelector(
    selectNavigationContainerDomain,
    substate => substate,
  );

const makeSelectTags = () =>
    createSelector(
        selectNavigationContainerDomain,
        substate => substate.tags,
    );

export default makeSelectNavigationContainer;
export { selectNavigationContainerDomain, makeSelectTags };
