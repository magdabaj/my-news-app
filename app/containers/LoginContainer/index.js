/**
 *
 * LoginContainer
 *
 */

import React, {useState} from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { compose } from "redux";

import { useInjectSaga } from "utils/injectSaga";
import { useInjectReducer } from "utils/injectReducer";
import makeSelectLoginContainer, {makeSelectLoginUser} from "./selectors";
import reducer from "./reducer";
import saga from "./saga";
import Login from "../../components/Login";
import {login, cancelLogin} from "./actions";

export function LoginContainer({...props}) {
  useInjectReducer({ key: "loginContainer", reducer });
  useInjectSaga({ key: "loginContainer", saga });

  // console.log({...props});


  return <div>
    <Login {...props} />
  </div>;
}

LoginContainer.propTypes = {
  dispatch: PropTypes.func.isRequired
};

const mapStateToProps = createStructuredSelector({
  loginContainer: makeSelectLoginContainer(),
  loginUser: makeSelectLoginUser(),
});

function mapDispatchToProps(dispatch) {
  return {
    login: (email, password) => dispatch(login(email, password)),
    cancelLogin: () => dispatch(cancelLogin()),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default compose(withConnect)(LoginContainer);
