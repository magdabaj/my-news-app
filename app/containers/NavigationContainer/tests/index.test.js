/**
 *
 * Tests for NavigationContainer
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from "react";
import { render } from "react-testing-library";
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { NavigationContainer } from "../index";
import {getTags, selectTag, toggleDrawer} from "../actions";
import {makeSelectIsDrawerOpen, makeSelectSelectedTag, makeSelectTags} from "../selectors";
import {makeSelectEmail} from "../../LoginContainer/selectors";
import {withStoreIntlAndRouter} from "../../../utils/testHelpers";

function renderNavigationContainer(args) {
  const defaultProps = {
    getTags: () => {},
    selectTag: () => {},
    toggleDrawer: () => {},
    tags: [],
    selectedTag: {},
    isDrawerOpen: false,
    email: '',
  };

  const props = {...defaultProps, ...args}

  return render(withStoreIntlAndRouter(<NavigationContainer {...props}/>))
}

describe("<NavigationContainer />", () => {
  it("Expect to not log errors in console", () => {
    const spy = jest.spyOn(global.console, "error");
    const dispatch = jest.fn();
    renderNavigationContainer();
    expect(spy).not.toHaveBeenCalled();
  });

  it("Expect to have additional unit tests specified", () => {
    expect(true).toEqual(true);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it("Should render and match the snapshot", () => {
    const {
      container: { firstChild }
    } = renderNavigationContainer();
    expect(firstChild).toMatchSnapshot();
  });
});
