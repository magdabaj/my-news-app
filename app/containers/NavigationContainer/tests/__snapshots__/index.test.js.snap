// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationContainer /> Should render and match the snapshot 1`] = `
<div>
  <div
    class="AppBar__AppBarContainer-c7n8kq-0 ciJhyx"
  >
    <div
      class="iconButton"
    >
      <span
        aria-hidden="true"
        class="fa fa-bars"
      />
    </div>
    <div
      class="Heading-zlrub7-0 hSXzTs"
    >
      <a
        class="StyledLink-sc-1rnvk1v-0 jWsYVy"
        href="/topics"
      >
        Coder daily
      </a>
    </div>
    <div
      class="LinkContainer-sc-1jaoxeg-0 dRHldZ"
    >
      <a
        href="/login"
      >
         login 
      </a>
    </div>
  </div>
  <div
    class="drawer"
  />
</div>
`;
